libgalois/include/galois/analytics/bfs/bfs.h

```diff
@@ -70,12 +68,17 @@ using BfsNodeDistance = galois::PODProperty<uint32_t>;
 /// not exist before the call.
 GALOIS_EXPORT Result<void> Bfs(
     graphs::PropertyFileGraph* pfg, size_t start_node,
-    const std::string& output_property_name,
-    BfsPlan algo = BfsPlan::Automatic());
+    const std::string& output_property_name, BfsPlan algo = {});
 
+/// Do a quick validation of the results of a BFS computation where the results
+/// are stored in property_name. This function does not do an exhaustive check.
+/// The results are approximate and make have false-negatives.
+/// @return True iff the results pass the validation checks, false if they do
+///   not, or an error resulting from the check algorithm itself.
```

Comment:
FWIW, I don't know what callers would behave differently between `false` from "had an error" here. Perhaps, `Result<void>`

Reply:
I see your point. We could go for a "throw on failure" function (It will literally raise an exception in Python). If so, I want to rename this to `BfsAssertValid` to make it clear that it's a check that will throw on failure. Thoughts?

Reply:
Raising an exception in the python API seems reasonable but the C++ code we have been writing hasn't been throwing exceptions so far. Is it the case that we are intending a strict or automated mirroring between the Python API and the C++ one?

If we do want to have an exception throwing version in the C++ library, I have a moderate preference to also have the `Result<void>` returning function too.

Reply:
Sorry. I was confusing. By raise exception, I just meant return an error object/code in C++. I see error codes as just a manual implementation of exceptions. The Python wrapper converts returns error objects into exceptions. I know we don't use C++ exceptions.

I still think it should be called `BfsAssertValid`. It converts a negative check into an error which is the fundamental action of an assertion.

Reply:
Cool. Got it.

`BfsAssertValid` seems reasonable to me.